### PR TITLE
luno-go: Adding interface to client to rate limit requests

### PR DESCRIPTION
**Pull request by donohutcheon**
Adding the ability to configure the client to rate limit requests.  This offers a better experience to API users who might otherwise send a burst of requests only to recieve a 429 error.

Refer to:
https://www.luno.com/en/developers/api#tag/Rate-Limiting

### CODE REVIEW

File: luno.go [hunk 1]
```diff
@@ -16,9 +16,14 @@ import (
 	"time"
 )
 
+type limiter interface {
```
Comment (neilgarb) [inline luno.go:19]: I think it's friendlier to export this.

File: README.md [hunk 2]
```diff
@@ -32,11 +32,14 @@ import (
   "context"
   "time"
   "github.com/luno/luno-go"
+  "golang.org/x/time/rate"
 )
 
 func main() {
+  rateLimiter := rate.NewLimiter(rate.Every(time.Minute/300), 1)
```
Comment (neilgarb) [inline README.md:39]: Maybe `rateLimiter := rate.NewLimiter(rate.Every(time.Minute), 300)` is more accurate?

Comment (donohutcheon) [reply README.md:39]: I tried initially.  What that allows is: a burst of 300 requests and then limits the requests to one per minute.

As I understand it, `rateLimiter := rate.NewLimiter(rate.Every(time.Minute/300), 1)` releases a token to the bucket every 0.2 seconds (60 seconds / 300) and only allows bursts of 1 request at a time.

**Comment by donohutcheon** [comment]
Adding a default rate limiter to the client.

File: luno.go [hunk 3]
```diff
@@ -14,26 +14,37 @@ import (
...
 	debug        bool
 }
 
-const defaultBaseURL = "https://api.luno.com"
-
-const defaultTimeout = 10 * time.Second
+const (
+	defaultBaseURL = "https://api.luno.com"
+	defaultTimeout = 10 * time.Second
+	defaultRate    = time.Minute / 300
```
Comment (neilgarb) [inline luno.go:38]: I think it would be useful to add a link to the API documentation where this limit is mentioned.

**Review by neilgarb — APPROVED** [review]

**Review by echarrod — APPROVED** [review]

**Review by neilgarb — DISMISSED** [review]

**Review by echarrod — DISMISSED** [review]

**Review by neilgarb — APPROVED** [review]

**Pull request merged**